game/yodesk.ksy

```diff
@@ -0,0 +1,1190 @@
+        type: u4
+        enum: puzzle_item_class
+        if: index != 0xFF_FF
+      - id: item2_class
+        type: u4
+        enum: puzzle_item_class
+        if: index != 0xFF_FF
+      - type: u2
+        if: index != 0xFF_FF
+      - id: strings
+        type: prefixed_str
+        repeat: expr
+        repeat-expr: 5
+        if: index != 0xFF_FF
+      - id: item_1
+        type: u2
+        if: index != 0xFF_FF
+      - id: item_2
+        type: u2
+        if: index != 0xFF_FF
+    enums:
+      puzzle_item_class:
+        0: keycard
+        1: tool
+        2: part
+        4: valuable
+        0xFFFF_FFFF: none
+  endf:
+    seq: []
+  characters:
+    seq:
+      - id: characters
+        type: character
+        repeat: until
+        repeat-until: _.index == 0xFF_FF
+  character:
+    seq:
+      - id: index
+        type: u2
+      - id: marker
+        contents: "ICHA"
+        if: index != 0xFF_FF
+      - id: size
+        type: u4
+        if: index != 0xFF_FF
+      - id: name
+        type: strz
+        size: 16
+        if: index != 0xFF_FF
+      - id: type
+        type: u2
+        enum: character_type
+        if: index != 0xFF_FF
+      - id: movement_type
+        type: u2
+        enum: movement_type
+        if: index != 0xFF_FF
+      - id: probably_garbage_1
+        type: u2
+        if: index != 0xFF_FF
+      - id: probably_garbage_2
+        type: u4
+        if: index != 0xFF_FF
+      - id: frame_1
+        type: char_frame
+        if: index != 0xFF_FF
+      - id: frame_2
+        type: char_frame
+        if: index != 0xFF_FF
+      - id: frame_3
+        type: char_frame
+        if: index != 0xFF_FF
+    enums:
+      character_type:
+        1: hero
+        2: enemy
+        4: weapon
+      movement_type:
+        0: none
+        4: sit
+        9: wander
+        10: patrol
+        12: animation
+  char_frame:
+    seq:
+      - id: tiles
+        type: u2
+        repeat: expr
+        repeat-expr: 0x8
+  character_auxiliaries:
+    seq:
+      - id: auxiliaries
+        type: character_auxiliary
+        repeat: until
+        repeat-until: _.index == 0xFF_FF
+  character_auxiliary:
+    seq:
+      - id: index
+        type: u2
+      - id: damage
+        type: s2
+        if: index != 0xFF_FF
+  character_weapons:
+    seq:
+      - id: weapons
+        type: character_weapon
+        repeat: until
+        repeat-until: _.index == 0xFF_FF
+  character_weapon:
+    seq:
+      - id: index
+        type: u2
+      - id: reference
+        doc: |
+          If the character referenced by index is a monster, this is a
+          reference to their weapon, otherwise this is the index of the
+          weapon's sound
+        type: u2
+        if: index != 0xFF_FF
+      - id: health
+        type: u2
+        if: index != 0xFF_FF
+  # Utilities
+  prefixed_str:
+    seq:
+      - id: len_content
+        type: u2
+      - id: content
+        type: str
+        size: len_content
```

Comment:
Please put `size` above `type` (https://doc.kaitai.io/ksy_style_guide.html#attr-id):

```suggestion
      - id: content
        size: len_content
        type: str
```

The same for all such `size:` occurrences in the file.